### ISSUE TITLE
 run the FARSITE in Windows.

### ISSUE THREAD

**Issue by niuniu238** [firelab/farsite#3]
I want to run the FARSITE in Windows.
 How can I do that?
thanks for your reading!
from a follower.
good luck!

**Comment by latwood** [comment]
I don't think this code will run on Windows, it's meant for a Linux system, and it doesn't have a GUI interface. In the past, I would use the Windows version of farsite from firelab.org whenever I needed to run farsite on Windows or with a GUI.
Just found out that in the last month or so that the windows version of farsite is not available anymore on firelab.org. However, flammap has a version of farsite inside of it so you can essentially run farsite on windows through flammap. I haven't played around with it, so I don't know if the user interface is what you would expect or as nice as the old windows farsite, but it does have all the farsite solver stuff. I do know that the old farsite program had a lot of specific help documentation that was only held internally to the program, and I have no clue if that got transferred over. There were some nice details on how the models worked in that old documentation.
If you have any major trouble running farsite through flammap on windows, I think Stuart Brittain should be able to help you.
Here are links to the firelab.org farsite and flammap information: https://www.firelab.org/project/farsite
https://www.firelab.org/project/flammap.

**Comment by niuniu238** [comment]
Thanks for your replying, well. I find that int main() of the code includes WIN32, which coulde run in Windows. And how can I convert the raw data to *.LCP or *.input file?
I want to run my example by using fule datas, terrian data , wearther datas and other sets, via the code.
发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

________________________________
发件人: latwood <notifications@github.com>
发送时间: Tuesday, September 17, 2019 11:49:04 AM
收件人: firelab/farsite <farsite@noreply.github.com>
抄送: niuniu238 <xvbenben@outlook.com>; Author <author@noreply.github.com>
主题: Re: [firelab/farsite] run the FARSITE in Windows. (#3)


I don't think this code will run on Windows, it's meant for a Linux system, and it doesn't have a GUI interface. In the past, I would use the Windows version of farsite from firelab.org whenever I needed to run farsite on Windows or with a GUI.
Just found out that in the last month or so that the windows version of farsite is not available anymore on firelab.org. However, flammap has a version of farsite inside of it so you can essentially run farsite on windows through flammap. I haven't played around with it, so I don't know if the user interface is what you would expect or as nice as the old windows farsite, but it does have all the farsite solver stuff. I do know that the old farsite program had a lot of specific help documentation that was only held internally to the program, and I have no clue if that got transferred over. There were some nice details on how the models worked in that old documentation.
If you have any major trouble running farsite through flammap on windows, I think Stuart Brittain should be able to help you.
Here are links to the firelab.org farsite and flammap information: https://www.firelab.org/project/farsite
https://www.firelab.org/project/flammap.

―

**Comment by latwood** [comment]
Honestly I usually ran WindNinja to get a fresh .lcp file and didnt need to
take the terrain data and fuel data to generate this from scratch. I know
the old farsite had a way where you could pick out these separate files and
it would stitch them together into a .lcp file, but I'm not sure if the
files were just text or binary.

There still aren't good details on what variables and what format to use
for the .input file, I put together this partial decipher when I was first
trying to use this code for Farsite:
https://github.com/firelab/farsite/blob/master/examples/test1177973.input.
I put this together by parsing through the old windows farsite help
documentation, then compared it with how this version of farsite parses the
inputs.
The examples also provide a few more useful input files specific to this
Linux code. The Windows version had similar inputs, and some kind of tools
where you could insert the raw data, and turn it into files specific for
Farsite for later, but I don't know if these tools exist in flammap.
In theory this code should work on Windows, and it has leftover stuff from
when it was originally taken from a Windows version, but not all the parts
are linked, there's a bunch of old dead code in there. Maybe the windows
stuff would work, but as far as I know, compiling the code in this repo has
not been tested on Windows.
If you really want to run this code on Windows but are okay with it not
being exactly Windows, what I do is use a virtual machine like VMware with
an iso of Ubuntu, or use the windows subsystem of Linux. Stuart Britain is
over the more up to date version of this code that only works on Windows
and not on Linux, but it isn't open to the public from what I can tell. At
some point in time we want to merge the two versions, but it's going to
take a lot of work. Its been discussed, but no specific timeline has been
discussed yet to start making this merge.

A lot of info, but I hope this helps.

**Comment by stubrittain** [comment]
FARSITE for Windows now resides in FlamMap6, and can be downloaded here:
https://www.firelab.org/project/flammap

For command line version of FARSITE for Windows that uses the same Farsite.dll as the above, you can download here:
https://www.alturassolutions.com/FB/FB_API.htm

The command line version contains sample data as well as documentation on the inputs file switches.

Also, using the GUI of FlamMap6 you can create Farsite runs and use the Run Log in the command line version as an inputs file.

You can create an LCP or GeoTIFF landscape in FlamMap6 from ascii grids. For creating other inputs you are either on your own or have a source for downloading the appropriate data.
